File: 287-find-the-duplicate-number/287-find-the-duplicate-number.java
class Solution {
    public int findDuplicate(int[] nums) {
        int n = nums.length;
        int l = 1, r = n - 1;
        while( l < r) {
            int m = (l + r) /2;
            int cnt = 0;
            for(int num : nums) {
                if(num >= l && num <= m) {
                    cnt++;
                }
            }
            if(cnt > m - l + 1) r = m;
            else l = m + 1;
        }
        return l;
    }
}

File: 287-find-the-duplicate-number/287-find-the-duplicate-number.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        nums.sort()
        prev = nums[0]
        for i in range(1, len(nums)):
            if(nums[i] == prev):
                return nums[i]
            else:
                prev = nums[i]
        return -1